src/connectors/stats/connectStats.js

```diff
@@ -88,5 +80,36 @@ export default function connectStats(renderFn, unmountFn = noop) {
     dispose() {
       unmountFn();
     },
+
+    getRenderState(renderState, renderOptions) {
+      return {
+        ...renderState,
+        stats: this.getWidgetRenderState(renderOptions),
+      };
+    },
+
+    getWidgetRenderState({ results, helper }) {
+      if (results) {
```

Comment:
Usually we try to reverse the flow and to remove unnecessary nesting:

```ts
// Let's get rid of this special flow
if (!results) {
  return /* ... */
}

// Let's treat the common case
return /* ... */
```

It's easier to follow the flow of the lifecycle.

Reply:
Fixed in https://github.com/algolia/instantsearch.js/pull/4565/commits/6d23d58b8408171da02dd73ff76dc2d68460fced :D